addon/build-waiter.ts

```diff
@@ -0,0 +1,37 @@
+import { DEBUG } from '@glimmer/env';
+import { TestWaiter } from 'ember-test-waiters';
+import NoopTestWaiter from './noop-test-waiter';
+
+/**
+ * Builds and returns a test waiter. The type of the
+ * returned waiter is dependent on whether the app or
+ * addon is in `DEBUG` mode or not.
+ *
+ * @param name {string} The name of the test waiter
+ * @returns {TestWaiter|NoopTestWaiter}
```

Comment:
I'd rather return an interface (instead of `SomeClass|OtherClass`), since both classes implement the same interface (`beginAsync` / `endAsync` / `register` / etc).

Reply:
Seems obvious in retrospect. For sure, will change.